src/Movim/Template/Builder.php

```diff
@@ -138,55 +139,31 @@ public function title(): string
      */
     public function language(): string
     {
-        $lang = $this->user && $this->user->language
-            ? $this->user->language
-            : Configuration::get()->locale;
-
-        if (empty($lang)) {
-            return 'en';
+        if ($this?->user?->language != null) {
+            $this->lang = $this->user->language;
+            return \Movim\i18n\Locale::printISO639($this->lang);
         }
 
-        // String casing for the web
-        $split = array_slice(preg_split('/[_-]/', $lang), 0, 3);
-        $lc = array_shift($split);
-        $langCode = strtolower($lc);
+        $l = \Movim\i18n\Locale::start();
 
-        if (empty($split)) {
-           return $langCode;
+        if (getenv('language') != null) {
+            $l->detect(getenv('language'));
+        } elseif (isset($_SERVER['HTTP_ACCEPT_LANGUAGE'])) {
+            $l->detect();
         }
 
-        $tags = array_map(function ($tag) {
-            switch (strlen($tag)) {
-                // Region
-                case 2:
-                case 3:
-                   return strtoupper($tag);
-                // Script
-                case 4:
-                   return ucfirst(strtolower($tag));
-                default:
-                   return $tag;
-            }
-        }, $split);
-
-        array_unshift($tags, $langCode);
-
-        return implode('-', $tags);
+        $this->lang = \Movim\i18n\Locale::printISO639(
+            $l?->language != null ? $l->language : Configuration::get()->locale,
+        );
+        return $this->lang;
```

Comment:
Simple syntax, but I like to put carriage return between block of code and before things like return.